qiskit/transpiler/passes/optimization/optimize_1q_commutation.py

```diff
@@ -126,23 +135,21 @@ def _commute_through(blocker, run, front=True):
                 commutation_rule = commutation_table[type(blocker.op)][preindex]
 
         if commutation_rule is not None:
-            while run_clone != []:
+            while len(run_clone) > 0:
                 next_gate = run_clone[0] if front else run_clone[-1]
                 if next_gate.name not in commutation_rule:
                     break
+                commuted.append(next_gate)
                 if front:
-                    commuted.append(next_gate)
-                    del run_clone[0]
+                    run_clone.popleft()
                 else:
-                    commuted.insert(0, next_gate)
-                    del run_clone[-1]
-
+                    run_clone.pop()
+        if not front:
+            commuted = commuted[::-1]
```

Comment:
I'm not sure reversing the list is faster than evaluating the 'if front` condition.

Reply:
I reversed the list to remove the `insert(0)` method in the original code. Previously an `insert` was being done at the first position in the `commuted` list and that was an `O(len(commuted))` operation. So as we iterate, the `insert` would be having quadratic complexity and that's why I changed that from the code. 
Maybe I'm missing something here, but that's what I thought initially.

Reply:
I reversed the list to remove the `insert(0)` method in the original code. Previously an `insert` was being done at the first position in the `commuted` list and that was an `O(len(commuted))` operation. So as we iterate, the `insert` would be having quadratic complexity and that's why I changed that from the code. 
Maybe I'm missing something here, but that's what I thought initially.

Reply:
Why not use a deque for `commuted` as well and then replace `insert(0, _)` with `appendleft`?